apps/extension/src/Approvals/Approvals.tsx

```diff
@@ -65,22 +70,36 @@ export const Approvals: React.FC = () => {
         />
         <Route
           path={TopLevelRoute.ApproveConnection}
-          element={<ApproveConnection />}
+          element={
+            <WithAuth>
+              <ApproveConnection />
+            </WithAuth>
+          }
         />
         <Route
           path={TopLevelRoute.ApproveDisconnection}
-          element={<ApproveDisconnection />}
+          element={
+            <WithAuth>
+              <ApproveDisconnection />
+            </WithAuth>
```

Comment:
I don't think it's possible to reach a state where the wallet is locked on this route (and the `WithAuth` instances that follow) in Namadillo, but it could be the case on another interface.